### PR TITLE
[AMORO-2145] Support Http interface for Tags and Branches

### LINKED ISSUE #2145: [Subtask]: Support Http interface for Tags and Branches
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

We have a requirement to display Tags and Branches on the Web UI, as mentioned in #2034 , so we need to introduce some HTTP interfaces for getting Tags and Branches, the interfaces can be like
- /tables/catalogs/{catalog}/dbs/{db}/tables/{table}/tags
- /tables/catalogs/{catalog}/dbs/{db}/tables/{table}/branches

### Parent issue

#1354

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by huyuanfeng2018**
## Why are the changes needed?

Close #2145.

## Brief change log

Two new interfaces are added to obtain the tag and branch information of the table.

- /tables/catalogs/{catalog}/dbs/{db}/tables/{table}/tags
- /tables/catalogs/{catalog}/dbs/{db}/tables/{table}/branches

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by wangtaohz — COMMENTED** [review]
Thanks for your work @huyuanfeng2018 👍 

I left some comments, please task a look.

**Comment by wangtaohz** [comment]
@huyuanfeng2018 There are some conflicts with the master branch, which were caused by Amoro introducing "spotless" and formatting a large amount of code. 

Once you have resolved the conflicts, you can execute `mvn spotless:apply` to format the code you modified in this PR.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/model/TagOrBranchInfo.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.dashboard.model;
+
+public class TagOrBranchInfo {
+  private String name;
```
[the remark this answers is not in the document]
Comment (huyuanfeng2018) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/model/TagOrBranchInfo.java:28]: done.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java [hunk 2]
```diff
@@ -285,6 +287,44 @@ public List<PartitionFileBaseInfo> getTableFiles(AmoroTable<?> amoroTable, Strin
...
 
+  @Override
+  public List<TagOrBranchInfo> getTableTags(AmoroTable<?> amoroTable) {
+    ArcticTable arcticTable = getTable(amoroTable);
+    List<TagOrBranchInfo> result = new ArrayList<>();
+    Map<String, SnapshotRef> snapshotRefs = getSnapshotRefs(arcticTable);
+    snapshotRefs.forEach(
+        (name, snapshotRef) -> {
+          if (snapshotRef.isTag()) {
+            result.add(new TagOrBranchInfo(
```
[the remark this answers is not in the document]
Comment (huyuanfeng2018) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java]: done.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java [hunk 3]
```diff
@@ -457,4 +497,18 @@ private void fillTableProperties(
   private ArcticTable getTable(AmoroTable<?> amoroTable) {
     return (ArcticTable) amoroTable.originalTable();
   }
+
+  private Map<String, SnapshotRef> getSnapshotRefs(ArcticTable table) {
+    if (table.isKeyedTable()) {
+      Map<String, SnapshotRef> baseTableRefs = table.asKeyedTable().baseTable().refs();
+      Map<String, SnapshotRef> changeTableRefs = table.asKeyedTable().changeTable().refs();
```
[the remark this answers is not in the document]
Comment (huyuanfeng2018) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java]: done.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java [hunk 4]
```diff
@@ -573,4 +587,23 @@ private void fillTableProperties(
...
     return (ArcticTable) amoroTable.originalTable();
   }
+
+  private List<TagOrBranchInfo> getTableTagsOrBranchs(
+      AmoroTable<?> amoroTable, Predicate<SnapshotRef> predicate) {
+    ArcticTable arcticTable = getTable(amoroTable);
+    List<TagOrBranchInfo> result = new ArrayList<>();
+    Map<String, SnapshotRef> snapshotRefs;
+    if (arcticTable.isKeyedTable()) {
+      return ImmutableList.of(TagOrBranchInfo.MAIN_BRANCH);
```
[the remark this answers is not in the document]
Comment (huyuanfeng2018) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java]: done.

**Review by wangtaohz — APPROVED** [review]
LGTM 👍

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/model/TagOrBranchInfo.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  public TagOrBranchInfo(String name, SnapshotRef snapshotRef) {
+    this.name = name;
+    this.snapshotId = snapshotRef.snapshotId();
+    this.minSnapshotsToKeep = snapshotRef.minSnapshotsToKeep();
+    this.maxSnapshotAgeMs = snapshotRef.maxSnapshotAgeMs();
+    this.maxRefAgeMs = snapshotRef.maxRefAgeMs();
+    if (snapshotRef.isTag()) {
+      this.type = "tag";
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/model/TagOrBranchInfo.java:58]: We should use constants(`public static final String xxxx`) to replace the using of "tag" and "branch".

Comment (huyuanfeng2018) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/model/TagOrBranchInfo.java]: done.
